File: adventofcode-2024/src/main/java/nl/mvdr/adventofcode/adventofcode2024/day02/Part1.java
package nl.mvdr.adventofcode.adventofcode2024.day02;

import java.util.stream.Stream;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import nl.mvdr.adventofcode.solver.LongSolver;

public class Part1 implements LongSolver {

    private static final Logger LOGGER = LoggerFactory.getLogger(Part1.class);

    @Override
    public long solve(Stream<String> lines) {
        return new RedNosedReportsSolver(false).solve(lines);
    }

    public static void main(String[] args) {
        var instance = new Part1();

        var result = instance.solve("input-day02-2024.txt");

        LOGGER.info(result);
    }
}
 

File: adventofcode-2025/src/main/kotlin/nl/mvdr/adventofcode/adventofcode2025/day02/Part1.kt
package nl.mvdr.adventofcode.adventofcode2025.day02

import io.github.oshai.kotlinlogging.KotlinLogging
import nl.mvdr.adventofcode.solver.FunctionSolver

private val logger = KotlinLogging.logger{}

fun solvePart1(lines: Sequence<String>): Long =
    lines.first()
        .split(",")
        .map { it -> it.split("-") }
        .map { it -> it.first().toLong() .. it.last().toLong() }
        .flatMap { it -> it.asSequence() }
        .filter(::isInvalid)
        .sum()

private fun isInvalid(id: Long): Boolean {
    val idString = id.toString()
    val length = idString.length
    return length % 2 == 0 && idString.take(length / 2) == idString.substring(length / 2)
}

fun main() {
    val result = FunctionSolver(::solvePart1).solve("input-day02.txt")
    logger.info { result }
}
